openedx/core/djangoapps/content/learning_sequences/api/outlines.py

```diff
@@ -85,21 +84,6 @@ def key_supports_outlines(opaque_key: OpaqueKey) -> bool:
     return False
 
 
-def public_api_available(course_key: CourseKey) -> bool:
-    """
-    Is the Public API available for this Course to this User?
-
-    This only really exists while we do the waffle-flag rollout of this feature,
-    so that in-process callers from other apps can determine whether they should
-    trust Learning Sequences API data for a particular user/course.
-    """
-    return (
-        key_supports_outlines(course_key) and
-        LearningContext.objects.filter(context_key=course_key).exists() and
```

Comment:
Do we have/need any error handling if we try to use the API for a course_key that doesn't support outlines/doesn't have a LearningContext object?

Reply:
My thinking is no? LearningContexts get generated at course publish time. I think this guard here was for the original rollout of this code.

But since then, everything on edx.org has been republished. And LS has been around for at least a full openedx release (maybe two?).

I'm going to note on Nutmeg's release page that LS is a new thing and if you haven't republished your courses for a couple releases, to do it now. (Though I assume that's standard practice for upgrading to an openedx release?)

Reply:
Small addendum - I just confirmed that learning contexts exist for everything in edx.org (except one old deleted course and old mongo courses).

Old mongo is a known and intentional gap for LS. But LS is only currently used in the MFE, which also doesn't support old mongo.